lib/gst/player/gstplayer.c

```diff
@@ -2641,6 +2658,46 @@ gst_player_seek_internal (gpointer user_data)
 }
 
 /**
+ * gst_player_set_rate:
+ * @player: #GstPlayer instance
+ * @rate: playback rate
+ *
+ * Playback at specified rate
+ */
+void
+gst_player_set_rate (GstPlayer *self, gdouble rate)
+{
+  g_return_if_fail (GST_IS_PLAYER (self));
```

Comment:
rate==0.0 is invalid and should be caught here, or we should convert rate==0.0 to pausing... but that then requires keeping track of much more state everywhere.